pkg/build/registry/buildlog/rest.go

```diff
@@ -43,62 +48,97 @@ func (r RealPodControl) getPod(namespace, name string) (*kapi.Pod, error) {
 func NewREST(b build.Registry, pn kclient.PodsNamespacer, connectionInfo kclient.ConnectionInfoGetter) *REST {
 	return &REST{
 		BuildRegistry:  b,
-		PodControl:     RealPodControl{pn},
+		PodGetter:      &podGetter{pn},
 		ConnectionInfo: connectionInfo,
+		Timeout:        defaultTimeout,
 	}
 }
 
-var _ = rest.Redirector(&REST{})
+var _ = rest.GetterWithOptions(&REST{})
 
-// Redirector implementation
-func (r *REST) ResourceLocation(ctx kapi.Context, id string) (*url.URL, http.RoundTripper, error) {
-	build, err := r.BuildRegistry.GetBuild(ctx, id)
+// Get returns a streamer resource with the contents of the build log
+func (r *REST) Get(ctx kapi.Context, name string, opts runtime.Object) (runtime.Object, error) {
+	buildLogOpts, ok := opts.(*api.BuildLogOptions)
+	if !ok {
+		return nil, errors.NewBadRequest("Did not get an expected options object.")
+	}
+	build, err := r.BuildRegistry.GetBuild(ctx, name)
 	if err != nil {
-		return nil, nil, fielderrors.NewFieldNotFound("Build", id)
+		return nil, errors.NewNotFound("Build", name)
 	}
-
-	// TODO: these must be status errors, not field errors
-	// TODO: choose a more appropriate "try again later" status code, like 202
+	switch build.Status {
```

Comment:
Will this switch handle a situation where build is being canceled when in New or Pending status?